src/api/app/jobs/project_do_project_copy_job.rb

```diff
@@ -1,7 +1,41 @@
 class ProjectDoProjectCopyJob < ApplicationJob
   queue_as :quick
 
+  attr_accessor :project, :params
   def perform(project_id, params)
-    Project.find(project_id).do_project_copy(params)
+    self.project = Project.find(project_id)
+    self.params = params
+    do_project_copy
+  end
+
+  private
+
+  def do_project_copy
+    User.find(params[:user]).run_as do
+      # copy entire project in the backend
+      begin
+        path = "/source/#{CGI.escape(project.name)}"
+        path << Backend::Connection.build_query_from_hash(params,
+                                                          [:cmd, :user, :comment, :oproject, :withbinaries, :withhistory,
+                                                           :makeolder, :makeoriginolder, :noservice])
+        Backend::Connection.post path
+      rescue Backend::Error => e
+        logger.debug "copy failed: #{e.summary}"
+        # we need to check results of backend in any case (also timeout error eg)
+      end
+      _update_backend_packages
+    end
+  end
+
+  def _update_backend_packages
+    # restore all package meta data objects in DB
+    backend_pkgs = Xmlhash.parse(Backend::Api::Search.packages_for_project(project.name))
+    backend_pkgs.elements('package') do |package|
+      pname = package['name']
+      p = project.packages.where(name: pname).first_or_initialize
+      p.update_from_xml(Xmlhash.parse(Backend::Api::Sources::Package.meta(project.name, pname)))
+      p.save! # do not store
```

Comment:
I just moved this code - but the comment is a little strange if you know
https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/package.rb#L68

Reply:
Guess it's older than the callback :-)

Reply:
the callback is from 2012 - at least :)

Reply:
any possibility to have a spec for that?

Reply:
Not with VCR

Reply:
don't get it. You can mock all backend calls, since we are not testing the backend, but a job class.

Reply:
Possibly. But this is tested in the minitest suite

Reply:
(as you can see with in the current failures :)